src/ProxyManager/FileLocator/FileLocator.php

```diff
@@ -24,10 +25,12 @@ public function __construct(string $proxiesDirectory)
         $absolutePath = realpath($proxiesDirectory);
 
         if ($absolutePath === false) {
-            throw InvalidProxyDirectoryException::proxyDirectoryNotFound($proxiesDirectory);
+            if (! is_dir($proxiesDirectory)) {
```

Comment:
Hmm, this is an added check: wasn't the problem the realpath call?

Reply:
Yes, but as you mentioned you also needed it to deal with symlinks, this seemed to be the better option, and would result in less change of behaviour.
is_dir will work with a stream wrapper, while realpath will not.

Reply:
Discussed in chat: I didn't see (during review) that `$proxiesDirectory` is no longer discarded right away